xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/ModelFactory.java

```diff
@@ -243,6 +249,7 @@ private void fillObjectSummary(ObjectSummary objectSummary, Document doc, BaseOb
                 objectSummary.setHeadline(serializePropertyValue(xwikiObject.get(propertyNames[0])));
             } catch (XWikiException e) {
                 // Should never happen
+                logger.error("Unexpected exception when accessing property [{}]", propertyNames[0], e);
```

Comment:
I never have seen this happening, but I wanted to log anything in case there is an error instead if just swallowing the exception. Then, level "warn" might be more appropriate in this case.

Reply:
I think error is better since it's not supposed to happen and thus if it does it would be quite unexpected and probably something to look at. My point was more that you left the comment untouched but added the logging and the 2 don't make sense. You need find what could happen in term of error and then remove "should never happen" and instead explain when it could happen for ex. Or mention that it shouldn't happen but that you don't know if it could happen and to be extra safe we're logging it as a way to know about it about analyze it to decide what to do in the future.

Reply:
ok, then I better skip that change, as I have no idea how the above lines could throw an exception,